build/commands/lib/build.js

```diff
@@ -20,22 +41,30 @@ const touchOverriddenFiles = () => {
 
   const chromiumSrcDir = path.join(config.srcDir, 'brave', 'chromium_src')
   var sourceFiles = util.walkSync(chromiumSrcDir, applyFileFilter)
+  const additionalGen = getAdditionalGenLocation()
 
   // Touch original files by updating mtime.
   const chromiumSrcDirLen = chromiumSrcDir.length
   sourceFiles.forEach(chromiumSrcFile => {
-    var overriddenFile = path.join(config.srcDir, chromiumSrcFile.slice(chromiumSrcDirLen))
-    if (!fs.existsSync(overriddenFile)) {
-      // Try to check that original file is in gen dir.
-      overriddenFile = path.join(config.outputDir, 'gen', chromiumSrcFile.slice(chromiumSrcDirLen))
+    const relativeChromiumSrcFile = chromiumSrcFile.slice(chromiumSrcDirLen)
+    var overriddenFile = path.join(config.srcDir, relativeChromiumSrcFile)
+    // If the original file doesn't exist, assume that it's in the gen dir.
+    const tryOverrideGen = !fs.existsSync(overriddenFile)
+    if (tryOverrideGen) {
+      overriddenFile = path.join(config.outputDir, 'gen', relativeChromiumSrcFile)
     }
 
     if (fs.existsSync(overriddenFile)) {
       // If overriddenFile is older than file in chromium_src, touch it to trigger rebuild.
-      if (fs.statSync(chromiumSrcFile).mtimeMs - fs.statSync(overriddenFile).mtimeMs > 0) {
-        const date = new Date()
-        fs.utimesSync(overriddenFile, date, date)
-        console.log(overriddenFile + ' is touched.')
+      updateFileUTimesIfOverrideIsNewer(overriddenFile, chromiumSrcFile)
+      // If the original file is in gen dir, then also check for secondary gen dir.
```

Comment:
Touching second gen is only done if gen has has a file with same name.
Same file are generated together on both gen dir?

Reply:
That seems to be the case.